app/src/rgb_underglow.c

```diff
@@ -346,6 +346,18 @@ int zmk_rgb_underglow_toggle() {
     return zmk_rgb_underglow_save_state();
 }
 
+int zmk_rgb_underglow_set_hsb(uint16_t hue, uint8_t saturation, uint8_t brightness) {
+    if (hue > 360 || saturation > 100 || brightness > 100) {
+        return -ENOTSUP;
+    }
+
+    state.hue = hue;
+    state.saturation = saturation;
+    state.brightness = brightness;
```

Comment:
Are we sure we don't want to set the animation to solid color? I guess this makes sense if you want to keep breathing. Staying on one of the spectrum animations seems weird though. Maybe it could be an option somehow. Kconfig doesn't feel like the place for it though.

Reply:
I waffled on this point for a bit and decided to leave it be ; if anything if you set the color to solid while a spectrum animation is running I'm inclined to say 'user error' or 'keymap bug'.

I don't know if this is tenable longer term but a config option would need to be through KConfig or similar `.conf` file and that seems like a kludge.

Reply:
Yeah this is fine as a start. Something for us to consider in the future.